src/main/scala/org/constellation/ConstellationNode.scala

```diff
@@ -26,15 +26,60 @@ import org.constellation.util.{APIClient, Metrics}
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-object ConstellationNode extends StrictLogging {
+// scopt requires default args for all properties.
+// Make sure to check for null early -- don't propogate nulls anywhere else.
+case class CliConfig(externalIp: java.net.InetAddress = null,
+                     externalPort: Int = 0,
+                     debug: Boolean = false)
 
-  val ConstellationVersion = "1.0.10"
+object ConstellationNode extends StrictLogging {
 
+  //noinspection ScalaStyle
   def main(args: Array[String]): Unit = {
     logger.info("Main init")
+
+    import scopt.OParser
+    val builder = OParser.builder[CliConfig]
+    val parser1 = {
+      import builder._
+      OParser.sequence(
+        programName("constellation"),
+        head("constellation", BuildInfo.version),
+        opt[java.net.InetAddress]("ip")
+          .action((x, c) => c.copy(externalIp = x))
+          .valueName("<ip address>")
+          .text("the ip you can be reached from outside"),
+        opt[Int]('p', "port")
+          .action((x, c) => c.copy(externalPort = x))
+          .text("the port you can be reached from outside"),
+        opt[Int]('d', "debug")
+          .action((x, c) => c.copy(externalPort = x))
+          .text("run the node in debug mode"),
+        help("help").text("prints this usage text"),
+        version("version").text(s"Constellation v${BuildInfo.version}"),
+        checkConfig(
+          c =>
+            if (c.externalIp != null && c.externalPort == 0 ||
+                c.externalPort != 0 && c.externalIp == null) {
+              failure("ip and port must either both be set, or neither.")
+            } else success
+        )
+      )
+    }
+
+    // OParser.parse returns Option[Config]
```

Comment:
So that comment wasn't updated -- but I think I get what you're coming from. Would be nice to extract this all to a class that dealt with merging cli args & args and the typesafe config file.

Furthermore -- I actually think we need a 2nd config file we don't have. I think the typesafe config file should only deal with tuning parameters and things like that, but things related to your crypto keys, ports, etc, should be separated to a new file.

I'll try and refactor this a bit before merging -- but I don't want to spend too much time on it atm so may not get it to an ideal place.